source/solvers/post_processors_smoothing.cc

```diff
@@ -236,29 +213,32 @@ QcriterionPostProcessorSmoothing<dim, VectorType>::generate_rhs(
   std::vector<Tensor<2, dim>>      present_velocity_gradients(n_q_points);
 
   // Fluid information
-  const FESystem<dim, dim> fe_fluid = dof_handler_fluid.get_fe();
-  FEValues<dim>            fe_values_fluid(*mapping_fluid,
-                                fe_fluid,
-                                quadrature_formula,
-                                update_values | update_quadrature_points |
-                                  update_JxW_values | update_gradients);
-
-  for (const auto &cell : dof_handler_fluid.active_cell_iterators())
+  const FESystem<dim, dim> fe_fluid = dof_handler_velocity.get_fe();
+  FEValues<dim>            fe_values_velocity(*mapping_fluid,
+                                   fe_fluid,
+                                   quadrature_formula,
+                                   update_quadrature_points | update_gradients);
+
+  for (const auto &cell : dof_handler_velocity.active_cell_iterators())
     {
       if (cell->is_locally_owned())
         {
-          fe_values_fluid.reinit(cell);
+          const auto &smoother_cell =
```

Comment:
This is because we are looping over the actives cells of two different dof handlers. So we cannot use the cell of one dof handler to reinit an feValues associated with the other dof handler.
I'll try to flesh out the comments a bit more.